lib/rubygems/specification.rb

```diff
@@ -2716,4 +2723,22 @@ def flatten_require_paths # :nodoc:
   def raw_require_paths # :nodoc:
     @require_paths
   end
+
+  # if we don't get the checksum from the server
+  # calculating the checksum from the file on disk still provides some measure of security
+  # if it changes from install to install, that is cause for concern
+  def to_checksum
+    return Bundler::Checksum.new(name, version, platform, "sha256-#{checksum}") if checksum
```

Comment:
is it safe for RubyGems to be referencing the just-introduced bundler constants?

Reply:
This is a good question. I don't know, I didn't realize that it could be a problem so let me know if I need to remove it. If so, I think we'll need to go back to the drawing board b/c there are a lot of specs that resolve to `Gem::Specification` and so if we can't put a checksum on this class then I'm not totally sure which direction to go

Reply:
We can customize/monkey-patch RubyGems classes in Bundler, but we do it in `lib/bundler/rubygems_ext.rb` to make them independent of the specific RubyGems version being installed.